curp/src/rpc/mod.rs

```diff
@@ -635,6 +639,36 @@ impl CurpError {
         Self::Internal(reason.into())
     }
 
+    /// Whether to abort fast round early
+    pub(crate) fn should_abort_fast_round(&self) -> bool {
+        matches!(
+            *self,
+            CurpError::Duplicated(_)
+                | CurpError::ShuttingDown(_)
+                | CurpError::InvalidConfig(_)
+                | CurpError::NodeAlreadyExists(_)
+                | CurpError::NodeNotExists(_)
+                | CurpError::LearnerNotCatchUp(_)
+                | CurpError::ExpiredClientId(_)
+                | CurpError::Redirect(_)
+        )
+    }
+
+    /// Whether to abort slow round early
+    pub(crate) fn should_abort_slow_round(&self) -> bool {
+        matches!(
+            *self,
+            CurpError::ShuttingDown(_)
+                | CurpError::InvalidConfig(_)
+                | CurpError::NodeAlreadyExists(_)
+                | CurpError::NodeNotExists(_)
+                | CurpError::LearnerNotCatchUp(_)
+                | CurpError::ExpiredClientId(_)
+                | CurpError::Redirect(_)
+                | CurpError::WrongClusterVersion(_)
+        )
+    }
+
```

Comment:
Should the fast round abort on `WrongClusterVersion`, and slow round abort on `Duplicated`?

Reply:
Perhaps your understanding and the naming of the function are just the opposite. Here, the meaning is that if a certain error occurs, it is necessary to terminate the fast/slow round in advance.

`slow_round` can continue execution after receiving `Duplicated`. It may retrieve the result from the cmd board before that has not been garbage collected yet.
Some offline servers may return `WrongClusterVersion`, but it may not affect the successful execution of the fast round. Therefore, `WrongClusterVersion` should not block the execution of the fast round.